pyproject.toml

```diff
@@ -33,8 +33,15 @@ dependencies = [
 ]
 [project.optional-dependencies]
 all = [
+  "asdf[lz4]",
+  "asdf[http]",
+]
+lz4 = [
```

Comment:
Although not required for this PR since I was changing `optional-dependencies` I figured I'd separate out the lz4 one so that downstream users (like roman_datamodels) could use `asdf[lz4]`.